retrofit2/src/main/java/com/linecorp/armeria/client/retrofit2/ArmeriaCallFactory.java

```diff
@@ -196,7 +186,18 @@ public Response execute() throws IOException {
 
         @Override
         public void enqueue(Callback callback) {
-            createRequest().subscribe(callFactory.subscriberFactory.create(this, callback, request));
+            if (httpResponse != null) {
+                throw new IllegalStateException("executed already");
+            }
+
+            // Run atomically
+            if (executionStateUpdater.compareAndSet(this, ExecutionState.IDLE, ExecutionState.RUNNING)) {
+                httpResponse = doCall(callFactory, request);
+            } else {
+                callback.onFailure(this, new IOException("Call state is not IDLE"));
```

Comment:
Since entering this block is semantically equivalent to L190

```suggestion
                throw new IllegalStateException("executed already");
```